src/content/blog/fastify.mdx

```diff
@@ -0,0 +1,179 @@
+codebases with tens of thousands of lines of code that rely on
+express and can't easily jump to an alternative.
+
+But Val Town's backend is not an enormous legacy project and switching
+web frameworks wasn't going to require
+a [deadly from-scratch rewrite][never_rewrite].
+
+Technology choice is complicated. I don't want to make a decision
+based on some simplistic criteria like GitHub Stars, but it's equally
+fraught to rely on vibes and instinct. So I investigated whether
+projects had good documentation, maintainership, if they had
+a decent history of stability, and perhaps most importantly, whether
+they were trying to solve the problems that we needed to solve.
+
+### Fastify
+
+We landed on [fastify][fastify] as the technology to build the
+future of Val Town's public API. Some of the strongest points for
+fastify were:
+
+- An extremely strong [ecosystem](https://fastify.dev/ecosystem/),
+  which includes a lot of high-quality, first-party plugins. We're
+  building a system in production, which means things like rate
+  limiting are going to be necessary, and it's amazing that
+  fastify's [rate limiter](https://github.com/fastify/fastify-rate-limit)
+  is well-architected.
+- Great, thorough [documentation](https://fastify.dev/docs/latest/Reference/Routes/),
+  and an active community that has been really helpful for support.
+- Solid support for [plugins and encapsulation](https://fastify.dev/docs/latest/Reference/Encapsulation/) that let us structure the application into sub-applications.
+- Support for [validation and serialization](https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/) with multiple validators and validation for all inputs and ouputs.
+- [Good support for async/await](https://fastify.dev/docs/latest/Reference/Routes/#async-await), like you might expect from a modern framework.
+- Off-the-shelf integrations for [opentelemetry](https://www.npmjs.com/package/@opentelemetry/instrumentation-fastify) and [Sentry](https://docs.sentry.io/platforms/javascript/guides/fastify/) mean
+  that we could have tracing and telemetry from day one.
+- Finally, [@fastify/express](https://github.com/fastify/fastify-express)
+  let us incrementally migrate from express to fastify.
+- We were already using the [light-my-request](https://github.com/fastify/light-my-request) module to test
+  our express server. light-my-request is written and maintained
+  by the fastify folks, and it is built into fastify.
+
+Of course, fastify is also very fast, but web framework overhead
+wasn't a problem with express and most likely won't be a problem with
+any framework we choose: most of our performance envelope has to
+do with operating [the Val Town runtime][runtimes] and interacting
+with the database.
+
+We considered a few other options, like [Hono](https://hono.dev/) and [Elysia](https://elysiajs.com/).
+Both are lovely projects - we use and recommend Hono a ton
```

Comment:
Probably a good place to do a shoutout to Hono and maybe even link out to a val that uses it. If this article has a "go use val town" component, I think this may be it